webapp/packages/core-app/src/NavigationTree/ElementsTree.tsx

```diff
@@ -42,30 +48,49 @@ interface Props {
   onOpen?: (node: NavNode) => Promise<void> | void;
   onSelect?: (node: NavNode, multiple: boolean) => void;
   isSelected?: (node: NavNode) => boolean;
+  onFilter?: (node: NavNode, value: string) => void;
 }
 
 export const ElementsTree: React.FC<Props> = observer(function ElementsTree({
   root,
   control,
+  localState,
   selectionTree = false,
   emptyPlaceholder,
   className,
   onOpen,
   onSelect,
   isSelected,
+  onFilter,
 }) {
   const nodeChildren = useChildren(root);
   const Placeholder = emptyPlaceholder;
+  const [state] = useState(() => new MetadataMap<string, INodeState>(() => ({ filter: '' })));
+
+  const elementsTreeState = localState || state;
+
+  const getMetadata = useCallback((node: NavNode) => elementsTreeState.get(node.id), [elementsTreeState]);
```

Comment:
`INodeState` & `elementsTreeState` & `localState` & `state` & `getMetadata`
all this names about Tree Node Metadata, but naming is too varius and confusing